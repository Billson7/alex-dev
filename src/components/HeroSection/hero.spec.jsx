import React from "react"
import { render, screen } from "@testing-library/react"
import { Hero } from "./"

describe("Hero Section", () => {

  test("renders Hero section correctly", () => {
    const { container } = render(<Hero />)
    
    expect(container).toMatchSnapshot()
  })


  test("renders Hero section heading", () => {
    render(<Hero />)
    
    expect(screen.getByTestId('profile-heading')).toBeInTheDocument();
  })

  test("renders Hero section caption", () => {
    render(<Hero />)
    
    expect(screen.getByTestId('profile-caption')).toBeInTheDocument();
  })

  test("renders Hero section intro", () => {
    render(<Hero />)
    
    expect(screen.getByTestId('profile-caption')).toBeInTheDocument();
  })

})
