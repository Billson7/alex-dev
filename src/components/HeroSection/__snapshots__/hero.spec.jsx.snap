// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Section renders Hero section correctly 1`] = `
<div>
  <section
    class="section section-hero"
  >
    <div
      class="flex-row"
    >
      <img
        alt="profile of Alex"
        class="profile-image"
        src="profile-picture.jpeg"
      />
      <h2
        class="profile-heading"
        data-testid="profile-heading"
      />
      <p
        class="profile-caption"
        data-testid="profile-caption"
      />
      <p
        class="profile-intro"
        data-testid="profile-intro"
      />
      <code>
        need a graphic here to push the other sections out of initial view
      </code>
    </div>
  </section>
</div>
`;
